providers/src/airflow/providers/apache/kafka/hooks/consume.py

```diff
@@ -23,6 +23,18 @@
 from airflow.providers.apache.kafka.hooks.base import KafkaBaseHook
 
 
+class KafkaAuthenticationError(Exception):
+    """Custom exception for Kafka authentication failures."""
+
+    pass
+
+
+def error_callback(err):
```

Comment:
Looks promising!

> `error_cb(kafka.KafkaError)`: Callback for generic/global error events

We have to filter only related one ( https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#pythonclient-kafkaerror ).

Reply:
Is this what you meant?

```
def error_callback(err):
    """Handle kafka errors."""
    if err.code() == KafkaError.NO_ERROR:
        return
    else:
        print("Exception received: ", err)
        raise KafkaAuthenticationError(f"Authentication failed: {err}")
```

Reply:
I think that we would like to catch only Authentication issue while error_callback consume all errors (https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#pythonclient-kafkaerror).

Reply:
This catches only `_AUTHENTICATION`.
```
def error_callback(err):
    """Handle kafka errors."""
    if err.code() == KafkaError._AUTHENTICATION:
        raise KafkaAuthenticationError(f"Authentication failed: {err}")
    print("Exception received: ", err)

```